### ISSUE TITLE
WebView2 ( History Clear Default Page Failing )

### ISSUE THREAD

**Issue by jarno9981** [MicrosoftEdge/WebView2Feedback#3153]
![image](https://user-images.githubusercontent.com/53493418/214082626-079bcc31-ceda-40ef-83ce-e014e6012d2e.png)

when using the latest webview2 pre-release or any below the default history page returns error when clicking the clear button

UWP - Latest
Microsoft.xaml.ui - latest

Windows 11 Latest beta

**Comment by jarno9981** [comment]
@champnic or anyone any idee why

**Comment by david-risney** [comment]
The edge://settings page (and many of the other built-in browser edge:// pages) is not supported in WebView2. You can see a [list of unsupported edge: URIs](https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/browser-features#blocked-internal-urls) in the docs.

**Comment by jarno9981** [comment]
I used egde://history to openpage but clearing gives error

**Comment by david-risney** [comment]
That's true. We shouldn't have an allowed edge: page try to use a disallowed edge: page. We should consider:
1. changing edge://history to not use edge://settings when clearing data when running in WebView2
2. changing edge://history to not have a clear data button when running in WebView2
3. disallowing edge://history when running in WebView2

What sort of app do you have and how do you use the edge://history page?

You may need to create your own mechanism to clear browsing data that is described in this [Clear Browsing Data document](https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/clear-browsing-data?tabs=csharp).

**Comment by jarno9981** [comment]
> That's true. We shouldn't have an allowed edge: page try to use a disallowed edge: page. We should consider:
> 1. changing edge://history to not use edge://settings when clearing data when running in WebView2
> 2. changing edge://history to not have a clear data button when running in WebView2
> 3. disallowing edge://history when running in WebView2
> 
> What sort of app do you have and how do you use the edge://history page?
> 
> You may need to create your own mechanism to clear browsing data that is described in this [Clear Browsing Data document](https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/clear-browsing-data?tabs=csharp).

On browser app we are writing with a team of 3 persons we want to implant history but i wanted the history in listview but webview2 database doesn't allow its self to be read because it locks after few times

**Comment by david-risney** [comment]
You might try using the WebView2 navigation events to keep track of history of where your user visits on your own since WebView2 doesn't provide APIs to access history data.

**Comment by jarno9981** [comment]
> You might try using the WebView2 navigation events to keep track of history of where your user visits on your own since WebView2 doesn't provide APIs to access history data.

Would there maybevcome api to access the history

**Comment by mikeduglas** [comment]
I see there is History file in the UDF/<profile> which is SQLite database. "Visits" table looks like a data source for History page.,

**Comment by jarno9981** [comment]
> I see there is History file in the UDF/profile which is SQLite database. "Visits" table looks like a data source for History page.,

So i can clear it with with sqlite qeury

**Comment by david-risney** [comment]
> I see there is History file in the UDF/profile which is SQLite database.

How history is stored is a chromium implementation detail that could change in the future. I wouldn't recommend taking a dependency on that for fear that something changes in the future and breaks your app.

> Would there maybevcome api to access the history

IIRC this wasn't on our list because its something the end developer should be able to implement on their own using navigation APIs. Is there an aspect of implementing history on top of WebView2 that doesn't work and would require WebView2 to expose a history API? You can open a new issue 'Feature Request: History APIs' if you'd like to ask for that.

**Comment by jarno9981** [comment]
> > I see there is History file in the UDF/profile which is SQLite database.
> 
> How history is stored is a chromium implementation detail that could change in the future. I wouldn't recommend taking a dependency on that for fear that something changes in the future and breaks your app.
> 
> > Would there maybevcome api to access the history
> 
> IIRC this wasn't on our list because its something the end developer should be able to implement on their own using navigation APIs. Is there an aspect of implementing history on top of WebView2 that doesn't work and would require WebView2 to expose a history API? You can open a new issue 'Feature Request: History APIs' if you'd like to ask for that.

I tryed making my costum history but i have problems with listview either it duplicates to items or you can only see 1 item for every search so if you search for example youtube.com and then youtube.com 15minutes later the last youtube search will not show in the db file
